Add Missing Devmode setting in Dev Guide Ubuntu Install Instructions

danintel opened the pull request:
For Sawtooth 1.2.x **nightly**, new settings are now required for the
Devmode consensus engine that was previously defaulted in 1.1 and before.
That is,
* sawtooth.consensus.algorithm.name=Devmode
* sawtooth.consensus.algorithm.version=0.1

Setting these values need to be documented in Step 3 of
docs/source/app_developers_guide/ubuntu.rst

These changes are for 1.2 nightly and are not required for 1.1 and earlier.

Signed-off-by: danintel <daniel.anderson@intel.com>

danintel commented:
FYI, I tested these steps on Ubuntu Bionic 18.04.1 LTS,  Sawtooth nightly 1.2.1.dev366.

```
$ sawtooth settings list
[2019-01-24 12:15:12.982 INFO     helpers] GET /state?address=000000 HTTP/1.1: 200 status, 1287 size, in 0.028207 s
sawtooth.consensus.algorithm.name: Devmode
sawtooth.consensus.algorithm.version: 0.1
sawtooth.settings.vote.authorized_keys: 0363a5584bd8e617e79dce964c52cc5c8a506b0...
monroe tmp $ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 18.04.1 LTS
Release:        18.04
Codename:       bionic
monroe tmp $ sawtooth -V
sawtooth-cli (Hyperledger Sawtooth) version 1.2.1.dev366
```

TheAsset25 commented:
I'm also having issues with the updated commands of "sudo -u sawtooth" ... when I try "sudo -u sawtooth sawset genesis -k /etc/sawtooth/keys/validator.priv" i get the response "Error: Unable to write to batch file: [Errno 13] Permission denied: 'config-genesis.batch'" (this is the same for any command with the updated 'sudo - sawtooth') ... i have to run 'sudo sawset ...' to get the above commands to run correctly. When I follow through with all of this it DOES seem to set the consensus protocol and version, but my validator is still not responding to successful POSTs to the REST API. At a bit of a loss on what to try. I also tried making sure the validator.toml is setup.

danintel commented:
FYI, the workaround for the "Unable to write to batch file" error is to be in a directory writable by user `sawtooth`.  For example: `cd /tmp`

TheAsset25 commented:
> FYI, the workaround for the "Unable to write to batch file" error is to be in a directory writable by user `sawtooth`. For example: `cd /tmp`

thank you!

danintel commented:
From the feedback above, I made two changes:
1. Switched Steps 3 and 4 (as the old Step 3 depended on the key generated in the old Step 4)
2. Added `cd /tmp` to avoid EPERM errors

danintel commented:
@TheAsset25  Regarding

> When I follow through with all of this it DOES seem to set the consensus protocol and version, but my validator is still not responding to successful POSTs to the REST API. At a bit of a loss on what to try. I also tried making sure the validator.toml is setup.

The other important step, with Sawtooth 1.1 and 1.2, is to start the consensus engine, preferably after starting `sawtooth-validator`. For the example documented here, start the Devmode consensus engine as follows:
`sudo -u sawtooth devmode-engine-rust -vv --connect tcp://localhost:5050`

The `--connect ...` option is required as the default in some Sawtooth 1.1 versions is incorrect (TCP port 5005 instead of the correct consensus engine port, 5050)

danintel commented:
Forced push to see if CI has been fixed recently.

danintel commented on docs/source/app_developers_guide/ubuntu.rst:
```diff
@@ -362,7 +373,9 @@ processor.
-         sawtooth.settings.vote.authorized_keys: 0276023d4f7323103db8d8683a4b7bc1eae1f66fbbf79c20a51185f589e2d304ce
+         sawtooth.consensus.algorithm.name: Devmode
+         sawtooth.consensus.algorithm.version: 0.1
+         sawtooth.settings.vote.authorized_keys: 0276023d4f7323103db8d8683a4b7bc1eae1f66...
```
Note that I changed the example output to be what the actual output now looks like (with a truncated key ending with `..`)

danintel commented:
CI is now finally working again. Could this trivial doc change, with the missing installation instructions, be approved and merged? Thanks!
